mlflow/entities/source_type.py

```diff
@@ -1,13 +1,14 @@
 class SourceType:
     """Enum for originating source of a :py:class:`mlflow.entities.Run`."""
 
-    NOTEBOOK, JOB, PROJECT, LOCAL, UNKNOWN = range(1, 6)
+    NOTEBOOK, JOB, PROJECT, LOCAL, UNKNOWN, PIPELINE = range(1, 7)
 
     _STRING_TO_SOURCETYPE = {
         "NOTEBOOK": NOTEBOOK,
         "JOB": JOB,
         "PROJECT": PROJECT,
         "LOCAL": LOCAL,
+        "PIPELINE": PIPELINE,
```

Comment:
Yes! That was my initial thought as well. But @dbczumar suggested that this is the standard that we have followed to set the sourceType of the run tags. Let me know what you think?

Reply:
Would prefer to set source type as an extension of this standard and add pipeline-specific tags over time. If we package individual steps as projects, the run should still be associated with type pipeline to indicate that it was produced via a pipeline execution (pipelines are not projects, and vice versa - project would be inaccurate).

Reply:
@sunishsheth2009 Can we set an additional `MLFLOW_PIPELINE_TEMPLATE_NAME` tag as part of this PR to keep track of the template name? `MLFLOW_PIPELINE_STEP` doesn't seem strictly necessary because we know that runs come from the train step of the regression pipeline & model versions come from the register step.